replication/gatys_ecker_bethge_2016/main.py

```diff
@@ -117,6 +117,67 @@ def figure_3(args):
         )
 
 
+def figure_2_author_information(args):
+    images = paper.images()
+    images.download(args.image_source_dir)
+
+    # Gatys provided this information
+    # compute layer_weights and use relu output -> impl_params=True
+    hyper_parameters = paper.hyper_parameters(impl_params=True)
+
+    hyper_parameters.nst.num_steps = 2000
+    hyper_parameters.nst.starting_point = "random"
+
+    content_image = images["neckarfront"].read(
+        size=hyper_parameters.nst.image_size, device=args.device
+    )
+
+    class StyleImage:
+        def __init__(self, label, image, weight_ratio):
+            self.label = label
+            self.image = image.read(
+                size=hyper_parameters.nst.image_size, device=args.device
+            )
+            self.weight_ratio = weight_ratio
+
+    style_images = (  # weight ratio from paper
+        StyleImage("B", images["shipwreck"], 1e-3),
+        StyleImage("C", images["starry_night"], 8e-4),
+        StyleImage("D", images["the_scream"], 5e-3),
+        StyleImage("E", images["femme_nue_assise"], 5e-4),
+        StyleImage("F", images["composition_vii"], 5e-4),
+    )
+
+    for style_image in style_images:
+        print(f"Replicating Figure 2 {style_image.label} with author information")
+        # The ratio is between a larger and a smaller value, according to the
+        # author the larger value was 1e6 or 1e9 (1e9 dont work)
+        larger_value = 1e6
```

Comment:
So here is still uncertainty and the author also couldn't give us the data he used for producing the results in the paper? Could you detail what doesn't work for `1e9`?

Reply:
As discussed offline, the author provided these two *discrete* values. We'll recheck `1e9`, but otherwise `1e6` is the "right" one.

Reply:
1. I have looked at the commentary again and the ratio here refers to the weight ratio given in the paper. For this, there must be a larger value and a smaller value, and according to the author, the larger value (weight for `style_loss`) was the discrete value `1e6` or `1e9`.
2. The optimisation is unstable with a weighting of 1e9. The following error message appears regularly:
```python
File "/home/julian/github/pystiche_papers/.venv/lib/python3.7/site-packages/torch/autograd/__init__.py", line 175, in backward
    allow_unreachable=True, accumulate_grad=True)  # Calls into the C++ engine to run the backward pass
RuntimeError: Function 'ConvolutionBackward0' returned nan values in its 1th output.`
```